diff --git a/browser/components/shell/nsKDEShellService.cpp b/browser/components/shell/nsKDEShellService.cpp
index 68d0d4465c54..cd4ec2854deb 100644
--- a/browser/components/shell/nsKDEShellService.cpp
+++ b/browser/components/shell/nsKDEShellService.cpp
@@ -101,3 +101,8 @@ nsKDEShellService::SetDesktopBackgroundColor(PRUint32 aColor)
     return NS_ERROR_NOT_IMPLEMENTED;
 }
 
+NS_IMETHODIMP
+nsKDEShellService::IsDefaultForScheme(nsTSubstring<char> const& aScheme, bool* aIsDefaultBrowser)
+{
+    return NS_ERROR_NOT_IMPLEMENTED;
+}
-- 
2.33.0

