diff --git a/browser/branding/branding-common.mozbuild b/browser/branding/branding-common.mozbuild
index f77058521ccb..26c4b0cbcdd3 100644
--- a/browser/branding/branding-common.mozbuild
+++ b/browser/branding/branding-common.mozbuild
@@ -41,24 +41,24 @@ def FirefoxBranding():
         ]
     elif 'gtk' in CONFIG['MOZ_WIDGET_TOOLKIT']:
         BRANDING_FILES += [
+            'default128.png',
             'default16.png',
             'default22.png',
             'default24.png',
+            'default256.png',
             'default32.png',
             'default48.png',
             'default64.png',
-            'default128.png',
-            'default256.png',
             'mozicon128.png',
         ]
         FINAL_TARGET_FILES.icons += ['mozicon128.png']
         FINAL_TARGET_FILES.chrome.icons.default += [
+            'default128.png',
             'default16.png',
             'default22.png',
             'default24.png',
+            'default256.png',
             'default32.png',
             'default48.png',
             'default64.png',
-            'default128.png',
-            'default256.png',
         ]
