diff --git a/browser/components/preferences/tests/browser_browser_languages_subdialog.js b/browser/components/preferences/tests/browser_browser_languages_subdialog.js
index 8b57bf08a83b..dd6f22094a93 100644
--- a/browser/components/preferences/tests/browser_browser_languages_subdialog.js
+++ b/browser/components/preferences/tests/browser_browser_languages_subdialog.js
@@ -13,7 +13,7 @@ const DICTIONARY_ID_PL = "pl@dictionaries.addons.mozilla.org";
 const TELEMETRY_CATEGORY = "intl.ui.browserLanguage";
 
 function langpackId(locale) {
-  return `langpack-${locale}@firefox.mozilla.org`;
+  return `langpack-${locale}@l10n.waterfox.net`;
 }
 
 function getManifestData(locale, version = "2.0") {
@@ -669,7 +669,7 @@ add_task(async function testInstallFromAMO() {
   is(getMainPaneLocales(), "en-US,pl,search", "en-US and pl now available");
 
   // Disable the Polish langpack.
-  langpack = await AddonManager.getAddonByID("langpack-pl@firefox.mozilla.org");
+  langpack = await AddonManager.getAddonByID("langpack-pl@l10n.waterfox.net");
   await langpack.disable();
 
   ({ dialogDoc, available, selected } = await openDialog(doc, true));
diff --git a/browser/locales/Makefile.in b/browser/locales/Makefile.in
index 1b8f21be1ca6..7ca5d391219a 100644
--- a/browser/locales/Makefile.in
+++ b/browser/locales/Makefile.in
@@ -21,9 +21,9 @@ PWD := $(CURDIR)
 ZIP_IN ?= $(ABS_DIST)/$(PACKAGE)
 
 ifdef MOZ_DEV_EDITION
-MOZ_LANGPACK_EID=langpack-$(AB_CD)@devedition.mozilla.org
+MOZ_LANGPACK_EID=langpack-$(AB_CD)@l10n.waterfox.net
 else
-MOZ_LANGPACK_EID=langpack-$(AB_CD)@firefox.mozilla.org
+MOZ_LANGPACK_EID=langpack-$(AB_CD)@l10n.waterfox.net
 endif
 # For Nightly, we know where to get the builds from to do local repacks
 ifdef NIGHTLY_BUILD
diff --git a/intl/locale/LangPackMatcher.sys.mjs b/intl/locale/LangPackMatcher.sys.mjs
index 977398b082e8..fd7c72939369 100644
--- a/intl/locale/LangPackMatcher.sys.mjs
+++ b/intl/locale/LangPackMatcher.sys.mjs
@@ -374,7 +374,7 @@ async function getAvailableLocales() {
   // If defaultLocale isn't lastFallbackLocale, then we still need the langpack
   // for lastFallbackLocale for it to be useful.
   if (defaultLocale != lastFallbackLocale) {
-    let lastFallbackId = `langpack-${lastFallbackLocale}@firefox.mozilla.org`;
+    let lastFallbackId = `langpack-${lastFallbackLocale}@l10n.waterfox.net`;
     let lastFallbackInstalled = await lazy.AddonManager.getAddonByID(
       lastFallbackId
     );
diff --git a/intl/locale/tests/LangPackMatcherTestUtils.sys.mjs b/intl/locale/tests/LangPackMatcherTestUtils.sys.mjs
index 4b18f1be134e..1e1008b55edd 100644
--- a/intl/locale/tests/LangPackMatcherTestUtils.sys.mjs
+++ b/intl/locale/tests/LangPackMatcherTestUtils.sys.mjs
@@ -38,7 +38,7 @@ export function getAddonAndLocalAPIsMocker(testScope, sandbox) {
         );
         resolve(
           availableLangpacks.map(locale => ({
-            guid: `langpack-${locale}@firefox.mozilla.org`,
+            guid: `langpack-${locale}@l10n.waterfox.net`,
             type: "language",
             hash: locale,
             target_locale: locale,
