From 889e53fcb19e12b31bc6ca7aec5958a60cd56f25 Mon Sep 17 00:00:00 2001
From: laniakea64 <laniakea64@users.noreply.github.com>
Date: Sun, 25 Mar 2018 14:34:53 -0400
Subject: [PATCH] Allow setting UTF-8 as fallback

---
 dom/encoding/FallbackEncoding.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/dom/encoding/FallbackEncoding.cpp b/dom/encoding/FallbackEncoding.cpp
index 6ecc12438395..3ed2d8a751b8 100644
--- a/dom/encoding/FallbackEncoding.cpp
+++ b/dom/encoding/FallbackEncoding.cpp
@@ -71,8 +71,7 @@ NotNull<const Encoding*> FallbackEncoding::Get() {
   // Don't let the user break things by setting the override to unreasonable
   // values via about:config
   auto encoding = Encoding::ForLabel(override);
-  if (!encoding || !encoding->IsAsciiCompatible() ||
-      encoding == UTF_8_ENCODING) {
+  if (!encoding || !encoding->IsAsciiCompatible()) {
     mFallback = nullptr;
   } else {
     mFallback = encoding;
