From: Mike Hommey <mh@glandium.org>
Date: Wed, 8 Apr 2020 08:43:20 +0900
Subject: Allow to build with older versions of nodejs 10

---
 python/mozbuild/mozbuild/nodeutil.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/python/mozbuild/mozbuild/nodeutil.py b/python/mozbuild/mozbuild/nodeutil.py
index 95bedbc..fef80cd 100644
--- a/python/mozbuild/mozbuild/nodeutil.py
+++ b/python/mozbuild/mozbuild/nodeutil.py
@@ -13,7 +13,7 @@ from mozboot.util import get_tools_dir
 from mozfile import which
 from six import PY3
 
-NODE_MIN_VERSION = StrictVersion("10.23.1")
+NODE_MIN_VERSION = StrictVersion("10.0")
 NPM_MIN_VERSION = StrictVersion("6.14.10")
 
 
