diff --git a/.mozconfig b/.mozconfig
index eeaf4f1..05b111b 100644
--- a/.mozconfig
+++ b/.mozconfig
@@ -39,8 +39,7 @@ ac_add_options --disable-tests
 ac_add_options --disable-verify-mar
 
 ac_add_options --enable-application=browser
-ac_add_options --enable-update-channel=release
-ac_add_options --enable-updater
+ac_add_options --disable-updater
 
 ac_add_options --with-app-name=waterfox
 ac_add_options --with-app-basename=Waterfox
