diff --git a/build/variables.py b/build/variables.py
index 86dfe508295e..ac8039f2b292 100644
--- a/build/variables.py
+++ b/build/variables.py
@@ -66,7 +66,7 @@ def get_info_from_sourcestamp(sourcestamp_path):
         return None, None
 
     # Return the repo and the changeset.
-    return lines[1].split('/rev/')
+    return lines[1].split('/tree/')
 
 
 def source_repo_header(output):
@@ -94,7 +94,7 @@ def source_repo_header(output):
         output.write('#define MOZ_SOURCE_STAMP %s\n' % changeset)
 
     if repo and buildconfig.substs.get('MOZ_INCLUDE_SOURCE_INFO'):
-        source = '%s/rev/%s' % (repo, changeset)
+        source = '%s/tree/%s' % (repo, changeset)
         output.write('#define MOZ_SOURCE_REPO %s\n' % repo)
         output.write('#define MOZ_SOURCE_URL %s\n' % source)
 
