diff --git a/waterfox/browser/branding/configure.sh b/waterfox/browser/branding/configure.sh
index 02163744fd711..34de2298dbf95 100644
--- a/waterfox/browser/branding/configure.sh
+++ b/waterfox/browser/branding/configure.sh
@@ -3,8 +3,6 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 MOZ_APP_DISPLAYNAME=Waterfox
-MOZ_APP_NAME=waterfox
-MOZ_DISTRIBUTION_ID=net.waterfox
 MOZ_INCLUDE_SOURCE_INFO=1
 MOZ_REQUIRE_SIGNING=
 MOZ_TELEMETRY_REPORTING=
